airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/models/metadata.py

```diff
@@ -0,0 +1,36 @@
+from metadata_service.models.generated.ConnectorMetadataDefinitionV1 import ConnectorMetadataDefinitionV1
+from pydantic import ValidationError
+
+from pydantic import ValidationError
+from typing import Tuple, Any, Optional
+
+
+class PydanticDelayValidationMixin:
```

Comment:
I find it's a breaking a bit the pydantic purpose if you delay to model validation.
I found an interesting suggestion [here](https://github.com/pydantic/pydantic/issues/668#issuecomment-512525178). Let me know what you think.

Reply:
@alafanechere Im glad you noticed this!

Its definately an awkward duck because we
1. Want strong type guarantees of what is metadata and what is not
2. Want these types to originate (be generated) from a jsonschema file to support java and python types
3. Want partial versions of these types to allow us to delay validation while we apply overrides

I went through a whole bunch of libraries / discussions on it.

Generally
1. Pydantic doesnt support partials out of the box
2. Pydantic doesnt seem to want to support this use case
3. The `pydantic-partial` library doesn't support nested validations 😢 

Ok so looking at the solution you linked. I like it, but I think theres a few small issues
1. Out of the box it doesn't give us access to the subscriptable methods __getitem__ and __setitem__
2. I actually tried to get this working but unfortunately theres its also not as easy as adding the subscriptable methods and we need to call some more internal apis to properly update the internal raw data dict (see screenshot below)
3. When looking at implementing this there were enough bugs I had to patch over that make me nervous to take at face value

In general I think im at the point where im happy to stay with the current implementation if it works. So long as you are 😅 

<img width="696" alt="image" src="https://user-images.githubusercontent.com/1325608/229927981-de04e566-58f8-4623-9162-c35306fe4f56.png">


### Appendix
https://github.com/team23/pydantic-partial
https://github.com/pydantic/pydantic/issues/1673
https://github.com/tiangolo/fastapi/issues/125

Reply:
Fair enough, let's keep current implementation!